### ISSUE TITLE
Adding custom parameter to retrieve access token by client credentials grant

### ISSUE THREAD

**Issue by JakSla** [scribejava/scribejava#996]
I am implementing machine2machine oauth2 token retrieval right now, for testing I wanted to use oauth0.com, but they need me to provide another param in a request (see example [here](https://skristof.medium.com/non-global-clients-are-not-allowed-access-to-apiv1-88877f1e855b): **_audience_**).

Using following method:
`        val token = oAuth20Service.getAccessTokenClientCredentialsGrant(scope)
`
Is there a way to add custom param 'audience' here?

There is a method 
`getAccessToken(AccessTokenRequestParams params)`
that takes params, but AFAIK it can handle only few predefined params.

tl;dr
is there a way to add custom parameter 'audience' to request sent while getting access token client credentials grant?

**Comment by iwesp** [comment]
This would also be very useful when trying to retrieve an access token from a system that expects a resource indicator (RFC 8707) to be present.

Currently, there seems to be no easy way to attach the required `resource` parameter.

**Comment by DaDom** [comment]
In case anyone is still looking for a solution: You can easily create your own extension of `OAuth20Service`:

```java
public class MyOAuth20Service extends OAuth20Service {

    private MyOAuth20Service(DefaultApi20 api, String apiKey, String apiSecret, String callback,
        String defaultScope, String responseType, OutputStream debugStream, String userAgent,
        HttpClientConfig httpClientConfig, HttpClient httpClient) {

        super(api, apiKey, apiSecret, callback, defaultScope, responseType, debugStream, userAgent, httpClientConfig,
            httpClient);
    }

    public OAuth2AccessToken fetchToken(OAuthRequest req) throws IOException, ExecutionException, InterruptedException {
        return this.sendAccessTokenRequestSync(req);
    }
}
```

You can then create a custom `OAuthRequest` with any additional parameter you need, and pass it to that service:

```java
String id = "your-client-id";
String secret = "your-client-secret";
OAuthRequest request = new OAuthRequest(Verb.POST, "your-token-url");
HttpBasicAuthenticationScheme.instance().addClientAuthentication(request, id, secret);
request.addParameter("grant_type", "client_credentials");
request.addParameter("client_id", id);
request.addParameter("scope", "your-scope");
request.addParameter("audience", "your-audience");
```
